backend/blog/models.py

```diff
@@ -32,11 +32,25 @@ def get_absolute_url(self):
         # TODO
         return ""
 
-    def save(self, *args, **kwargs):
-        if BlogPost.objects.filter(title=self.title).exists():
+
+class BlogPostAdminForm(forms.ModelForm):
+    class Meta:
+        model = BlogPost
+        fields = ['author', 'title', 'subtitle', 'slug', 'content', 'tags', 'published',
+                  'featured']
+
+    def clean(self):
+
+        title = self.cleaned_data["title"]
+        slug = self.cleaned_data['slug']
+
+        if BlogPost.objects.filter(title=title).exists():
             date = datetime.datetime.today()
-            self.slug = '%s-%i-%i-%i' % (slugify(self.title), date.year, date.month,
-                                         date.day)
-        elif not self.slug:
-            self.slug = slugify(self.title)
-        super(BlogPost, self).save(*args, **kwargs)
+            self.cleaned_data['slug'] = '%s-%i-%i-%i' % (slug, date.year, date.month,
```

Comment:
Hey — this is almost there but the logic isn't sound.
I have two blog posts, both called "The Blogpost". 
Post A gets created fine, the slug is "the-blogpost"
Post B errors out on creation (great!) and I manually set "the-blogpost-1" and hit save.
Instead, the slug is overwritten as "the-blogpost-1-[today's date]"
We should not be overwriting the user's unique set slug!